tasks/createTagsTasks.ts

```diff
@@ -0,0 +1,169 @@
+        logError('Failed to find roslyn commit.');
+        return;
+    }
+
+    const tagCreatedInRoslyn = await tagRepoAsync(
+        'dotnet',
+        'roslyn',
+        roslynCommit,
+        `VSCode-CSharp-${options.releaseVersion}`,
+        `${options.releaseVersion} VSCode C# extension release`
+    );
+
+    if (!tagCreatedInRoslyn) {
+        logError('Failed to tag roslyn');
+        return;
+    }
+
+    console.log('tag created in roslyn');
+
+    const tagCreatedInVsCodeCsharp = await tagRepoAsync(
+        'dotnet',
+        'vscode-csharp',
+        options.releaseCommit,
+        options.releaseVersion,
+        options.releaseVersion
+    );
+
+    if (!tagCreatedInVsCodeCsharp) {
+        logError('Failed to tag vscode-csharp');
+        return;
+    }
+
+    console.log('tag created in vscode-csharp');
+});
+
+async function findRoslynCommitAsync(): Promise<string | null> {
+    const packageJsonString = fs.readFileSync('./package.json').toString();
+    const packageJson = JSON.parse(packageJsonString);
+    const roslynVersion = packageJson['defaults']['roslyn'];
+    if (!roslynVersion) {
+        logError("Can't find roslyn version in package.json");
+        return null;
+    }
+
+    console.log(`Roslyn version is ${roslynVersion}`);
+    const nugetFileString = fs.readFileSync('./server/NuGet.config').toString();
+    const nugetConfig = await xml2js.parseStringPromise(nugetFileString);
+    const nugetServerIndex = nugetConfig.configuration.packageSources[0].add[0].$.value;
+    if (!nugetServerIndex) {
+        logError("Can't find nuget server index.");
+        return null;
+    }
+
+    console.log(`Nuget server index is ${nugetServerIndex}`);
+    // Find the matching commit from nuget server
+    // 1. Ask the PackageBaseAddress from nuget server index.
+    // 2. Download .nuspec file. See https://learn.microsoft.com/en-us/nuget/api/package-base-address-resource#download-package-manifest-nuspec
+    // 3. Find the commit from .nuspec file
+    const indexResponse = await sendHttpsGetRequestAsync(nugetServerIndex);
+    if (!indexResponse) {
+        return null;
+    }
+
+    const resources = indexResponse.data['resources'] as any[];
+    // This is required to be implemented in nuget server.
+    const packageBaseAddress = resources.find((res, _) => res['@type'] === 'PackageBaseAddress/3.0.0');
+    const id = packageBaseAddress['@id'];
+    console.log(`packageBaseAddress is: ${id}`);
+
+    const lowerCaseLanguageServerPackageName = 'Microsoft.CodeAnalysis.LanguageServer'.toLocaleLowerCase();
+    const nuspecGetUrl = `${id}${lowerCaseLanguageServerPackageName}/${roslynVersion}/${lowerCaseLanguageServerPackageName}.nuspec`;
+    console.log(`Url to get nuspec file is ${nuspecGetUrl}`);
+    const nuspecResponse = await sendHttpsGetRequestAsync(nuspecGetUrl);
+    if (!nuspecResponse) {
+        return null;
+    }
+
+    const nuspecFile = await xml2js.parseStringPromise(nuspecResponse.data);
```

Comment:
Instead of adding this whole dependency, we can just use a regex to find the correct line - I've done this for an xml case already here for example - https://github.com/dotnet/vscode-csharp/blob/main/src/lsptoolshost/dotnetRuntimeExtensionResolver.ts#L196

Reply:
Done